backend/src/middleware/notifications/mentions-in-grops.spec.ts

```diff
@@ -0,0 +1,845 @@
+  })
+
+  describe('post in public group', () => {
+    beforeAll(async () => {
+      jest.clearAllMocks()
+      authenticatedUser = await groupMember.toJson()
+      await markAllAsRead()
+      authenticatedUser = await postAuthor.toJson()
+      await markAllAsRead()
+      await mutate({
+        mutation: createPostMutation,
+        variables: {
+          id: 'public-post',
+          title: 'This is the post in the public group',
+          content: `Hey ${mentionString}! Please read this`,
+          groupId: 'public-group',
+        },
+      })
+    })
+
+    it('sends a notification to the no member', async () => {
+      authenticatedUser = await noMember.toJson()
+      await expect(
+        query({
+          query: notificationQuery,
+          variables: {
+            read: false,
+          },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          notifications: [
+            {
+              from: {
+                __typename: 'Post',
+                id: 'public-post',
+              },
+              read: false,
+              reason: 'mentioned_in_post',
+            },
+          ],
+        },
+        errors: undefined,
+      })
+    })
+
+    it('sends a notification to the pending member', async () => {
+      authenticatedUser = await pendingMember.toJson()
+      await expect(
+        query({
+          query: notificationQuery,
+          variables: {
+            read: false,
+          },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          notifications: [
+            {
+              from: {
+                __typename: 'Post',
+                id: 'public-post',
+              },
+              read: false,
+              reason: 'post_in_group',
+            },
+            {
+              from: {
+                __typename: 'Post',
+                id: 'public-post',
+              },
+              read: false,
+              reason: 'mentioned_in_post',
+            },
+          ],
+        },
+        errors: undefined,
+      })
+    })
+
+    it('sends a notification to the group member', async () => {
+      authenticatedUser = await groupMember.toJson()
+      await expect(
+        query({
+          query: notificationQuery,
+          variables: {
+            read: false,
+          },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          notifications: [
+            {
+              from: {
+                __typename: 'Post',
+                id: 'public-post',
+              },
+              read: false,
+              reason: 'post_in_group',
+            },
+            {
+              from: {
+                __typename: 'Post',
+                id: 'public-post',
+              },
+              read: false,
+              reason: 'mentioned_in_post',
+            },
+          ],
+        },
+        errors: undefined,
+      })
+    })
+
+    it('sends 5 emails, three mentions and 2 post in group', () => {
```

Comment:
context is completley missing

Should send only 1 email? I am unsure

Reply:
i think its bad practice to write tests for behaviour we actually want to prevent, but for the sake of the separation of different PR parts, I think it is acceptable.